src/Compilers/CSharp/Portable/Symbols/Source/SourceOrdinaryMethodSymbol.cs

```diff
@@ -697,24 +700,46 @@ private static void PartialMethodChecks(SourceOrdinaryMethodSymbol definition, S
 
             PartialMethodConstraintsChecks(definition, implementation, diagnostics);
 
-            SourceMemberContainerTypeSymbol.CheckValidNullableMethodOverride(
+            if (definition.Parameters.Length == implementation.Parameters.Length)
```

Comment:
>if [](http://example.com/codeflow?start=12&length=2)

Consider making this check part of the final `if` in the method, to avoid reporting the same error twice if the partial methods differ by parameter type and name, and also to have a single place in the method where we report this particular diagnostic.